R/check_forecasts.R

```diff
@@ -0,0 +1,267 @@
+                      "This can't be right and needs to be resolved. Maybe you need to check",
+                      "the unit of a single forecast and add missing columns?"))
+    check[["duplicate_forecasts"]] <- data[InternalDuplicateCheck > 1]
+  }
+  data[, InternalDuplicateCheck := NULL]
+
+  # check whether there is a model column present. And if not, state what that means
+  if (!("model" %in% colnames(data))) {
+    msg <- c(msg,
+             paste("There is no column called `model` in the data.",
+                   "scoringutils therefore thinks that all forecasts come from the same model"))
+    data[, model := "Unspecified model"]
+
+  }
+
+  # some checks whether there are the same number of quantiles, samples
+  data[, InternalNumCheck := length(prediction), by = obs_unit]
+  n <- unique(data$InternalNumCheck)
+  if (length(n) > 1) {
+    warnings <- c(warnings,
+                  paste0("Some forecasts have different numbers of rows (e.g. quantiles or samples). ",
+                         "scoringutils found: ", paste(n, collapse = ", "),
+                         ". This is not necessarily a problem, but make sure this is intended."))
+  }
+  check[["rows_per_forecast"]] <-
+    data[, .(rows_per_forecast = unique(InternalNumCheck)), by = model]
+  data[, InternalNumCheck := NULL]
+
+  # get available unique values per model for the different columns
+  cols <- obs_unit[!(obs_unit == "model")]
+  check[["unique_values"]] <-
+    data[, lapply(.SD, FUN = function(x) length(unique(x))), by = "model"]
+
+  check[["messages"]] <- unlist(msg)
+  check[["warnings"]] <- unlist(warnings)
+  check[["errors"]] <- unlist(errors)
+
+  attr(check, "class") <- "scoringutils_check"
+
+  return(check)
+}
+
+
+#' @title Print output from `check_forecasts()`
+#'
+#' @description Helper function that prints the output generated by
+#' [check_forecasts()]
+#'
+#' @param x An object of class 'scoringutils_check' as produced by
+#' [check_forecasts()]
+#' @param ... additional arguments (not used here)
+#'
+#' @return NULL
+#' @method print scoringutils_check
+#' @export
+
+print.scoringutils_check <- function(x, ...) {
+
+  print_elements <- names(x)[!(names(x) %in% c("messages"))]
+  print.default(x[print_elements])
+
+  cat(paste0("\nBased on your input, scoringutils thinks:\n",
+             paste(x$messages, collapse = "\n")))
+  cat("\n$rows_per_forecast shows how many rows (usually quantiles or samples are available per forecast.")
+  cat("\n$unique_values shows how many unique values there are per column per model",
+      "(across the entire data).")
+
+  if (length(x$warnings) > 0) {
+    cat(paste0("\n\n",
+               "You should be aware of the following warnings:\n",
+               paste(x$warnings, collapse = "\n")))
+  }
+
+  if (length(x$errors) > 0) {
+    cat(paste0("\n\n",
+               "The following things will likely result in an error:",
+               paste(x$errors, collapse = "\n")))
+  }
+  return(NULL)
```

Comment:
```suggestion
  return(invisible(x))
```

To match the behaviour of base methods

Reply:
:+1: